Matveibos/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# this class is Hash
+class MyHash
+  include Enumerable
+
+  attr_reader :pair_count, :arr
+
+  LOAD_FACTOR = 0.7
+  INITIAL_SIZE = 10
+  FOUR = 4
+  def initialize
+    initialize_hash
+  end
+
+  def [](key)
+    ident = key_hash(key)
+    arr_find(ident, key)&.last
+  end
+
+  def []=(key, value)
+    resize if pair_count > @size * LOAD_FACTOR
+    ident = key_hash(key)
+    pair = arr_find(ident, key)
+    if pair.nil?
```

Comment:
What do you think if we will rotate this conditional expression a little bit? `arr_find` will return you `nil` or array element, which is a truthful value. So let's check if array element is present first, and if not, push new value to our hash. With this change we will not call `.nil?` method one more time, which is apparently not needed here